src/DynamoCore/Graph/Annotations/AnnotationModel.cs

```diff
@@ -498,9 +500,19 @@ internal void UpdateBoundaryFromSelection()
 
                 this.X = region.X;              
                 this.Y = region.Y;
-                this.Width = Math.Max(region.Width, TextMaxWidth + ExtendSize);
                 this.ModelAreaHeight = IsExpanded ? region.Height : ModelAreaHeight;
-                this.Height = this.ModelAreaHeight + TextBlockHeight;
+                Height = this.ModelAreaHeight + TextBlockHeight;
+
+                if (IsExpanded)
+                {
+                    Width = Math.Max(region.Width, TextMaxWidth + ExtendSize);                    
+                    lastExpandedWidth = Width;
+                }
+                else
+                {
+                    //If the annotation is not expanded, than it will remain the same width of the last time it was expanded
+                    Width = lastExpandedWidth;
```

Comment:
Would there be any edge case that the new width should not be set to last expanded width?

Reply:
Thinking about it, maybe no edge case since we currently disable adding new elements into group when it is collapsed

Reply:
That's right, the width will be the same when the group is collapsed.